env/src/main/java/com/sap/cloud/security/config/k8s/K8sEnvironment.java

```diff
@@ -48,15 +55,58 @@ public Type getType() {
 		return Type.KUBERNETES;
 	}
 
+	private void loadAll() {
+		Map<String, OAuth2ServiceConfiguration> xsuaaPlans = DefaultServiceBindingAccessor.getInstance().getServiceBindings().stream()
+				.filter(b -> XSUAA.getCFName().equalsIgnoreCase(b.getServiceName().orElse(null)))
+				.map(b -> mapToOAuth2ServiceConfiguration(b)).collect(Collectors.toMap(c -> c.getProperty("plan"),
+						Function.identity()));
+		Map<String, OAuth2ServiceConfiguration> identityPlans = DefaultServiceBindingAccessor.getInstance().getServiceBindings().stream()
+				.filter(b -> IAS.getCFName().equalsIgnoreCase(b.getServiceName().orElse(null)))
+				.map(b -> mapToOAuth2ServiceConfiguration(b)).collect(Collectors.toMap(c -> c.getProperty("plan"),
+						Function.identity()));
```

Comment:
You could consider not loading the bindings twice:

```suggestion
		List<ServiceBinding> serviceBindings = DefaultServiceBindingAccessor.getInstance().getServiceBindings();
		Map<String, OAuth2ServiceConfiguration> xsuaaPlans = serviceBindings.stream()
				.filter(b -> XSUAA.getCFName().equalsIgnoreCase(b.getServiceName().orElse(null)))
				.map(this::mapToOAuth2ServiceConfiguration).collect(Collectors.toMap(c -> c.getProperty("plan"),
						Function.identity()));
		Map<String, OAuth2ServiceConfiguration> identityPlans = serviceBindings.stream()
				.filter(b -> IAS.getCFName().equalsIgnoreCase(b.getServiceName().orElse(null)))
				.map(this::mapToOAuth2ServiceConfiguration).collect(Collectors.toMap(c -> c.getProperty("plan"),
						Function.identity()));
```